src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const split = fromFormat[3];
+  const dayOfYear = date.split(split);
+  const tab = [];
+
+  for (let i = 0; i < 3; i++) {
+    for (let j = 0; j < 3; j++) {
+      if (fromFormat[i] === toFormat[j]) {
+        tab[j] = dayOfYear[i];
+      }
+    }
+  }
```

Comment:
The nested loops are unnecessary and inefficient for mapping the date parts from the old format to the new one. You could use an object to map the indices directly, which would simplify the logic and improve performance.